### PR TITLE
Improve dexterity support and optional transcoding

### PR DESCRIPTION
**Pull request by erral**
I have made some changes to improve the dexterity support: 
- there were an error due to the use of getField to get the information of the files
- now dexterity files are uploaded automatically to S3, instead of requiring a manual action.

I also included an option in the control panel to make the video transcoding optional.

I also documented a bit the use of the transcoder and the creation of the needed bucket for transcoded files.

I also added an example file to configure a redis instance using buildout.

### CODE REVIEW

**Comment by mattss** [comment]
Thanks for this @erral  - some very useful additions! Will review the pull request and get back to you.

File: src/netsight/cloudstorage/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from Products.CMFCore.utils import getToolByName
+
+
+PROFILE_ID = 'profile-netsight.cloudstorage:default'
+
+
+def upgrade_to_0002(context, logger=None):
+    if logger is None:
+        from logging import getLogger
+        logger = getLogger('esdrt.content.upgrades.0001_0002')
```
Comment (mattss) [inline src/netsight/cloudstorage/upgrade.py]: This looks like a copy-paste error in the logger name. Should be netsight.cloudstorage.upgrades or similar

File: README.rst [hunk 2]
```diff
@@ -114,12 +121,25 @@ by passing 'transcoded=true' on the file download request e.g.
 
 Files are currently transcoded using the 'Generic 480p 16:9' preset (`1351620000001-000020`_)
 
+To enable transcoding, you first need to create a specific S3 bucket to save the
+transcoded files. This specific S3 bucket must be called like the one used to
+store the files, but ended in "-transcoded". If you S3 bucket will be called
```
Comment (mattss) [inline README.rst]: I think this should ready 'If your S3 bucket is called...'

**Comment by erral** [comment]
Indeed :)

I just updated them.

**Comment by wengole** [comment]
@erral This looks good. I'll find some time next week to check it out and run some manual tests.

Maybe I'll eventually get around to writing some actual tests

**Comment by wengole** [comment]
@erral @mattss I reviewed this in a local instance on Tuesday. Seems to work fine. Optional transcoding optionally transcodes :-)

**Comment by erral** [comment]
:+1:

**Pull request merged**